apps/evm/lib/evm/message_call.ex

```diff
@@ -64,21 +88,32 @@ defmodule EVM.MessageCall do
   end
 
   defp execute(message_call) do
-    # first transitional state
-    message_call = transfer_value_to_recipient(message_call)
-
     message_call
+    |> transfer_funds_if_needed()
     |> prepare_call_execution_env()
     |> execute_call(message_call)
     |> update_state(message_call)
   end
 
+  defp transfer_funds_if_needed(message_call) do
+    if sender_is_recipient?(message_call) do
```

Comment:
if [value is `0`](https://github.com/poanetwork/mana/pull/331/files#diff-ca8e96a2417500c71d9313d86cffcbb3R140) don't you no longer need the conditional? `deletegatecall` will just transfer 0 which should do nothing right?

Reply:
Yeah, I was adding this for completeness since it has that condition in the Yellow Paper. But I think I'm going to split that into a separate issue if I add it at all. I'm not sure if this is breaking the build.